website/compatibility.md

```diff
@@ -0,0 +1,36 @@
+---
+layout: default
+title: Compatibility matrix
+nav_order: 2
+---
+
+# Compatibility matrix
+
+{% comment %}
+XXX: Once available on the default branch, include a link to the
+`generate-version-compatibility-overview.sh` script.
+{% endcomment %}
+
+Error Prone Support releases are generally compatible with only a limited
+number of Error Prone releases. The table below shows, for each Error Prone
+Support release, the Error Prone versions it is expected to be compatible with.
+Compatibility is determined by:
+1. Compiling and testing the Error Prone Support release source code against a
+   given Error Prone version. This validates source and behavioral
+   compatibility.[^1] [^2]
+2. Applying the released Refaster rules using a given Error Prone version. This
+   validates that the rules can be read by the targeted version of Error Prone,
+   proving that the serialization format is compatible.
+
+| Error Prone Support version | Compatible Error Prone versions |
+| --------------------------- | ------------------------------- |
+{% for release in site.data.compatibility.releases -%}
+| [{{ release.version }}](https://github.com/PicnicSupermarket/error-prone-support/releases/tag/v{{ release.version }}) | {%
+for version in release.compatible -%}
+[{{ version }}](https://github.com/google/error-prone/releases/tag/v{{ version }}){% unless forloop.last %}, {% endunless %}
+{%- endfor %} |
```

Comment:
The formatting is funny here, but that seemed necessary to make sure that a proper Markdown table is produced. Suggestions welcome.

Reply:
![image](https://github.com/PicnicSupermarket/error-prone-support/assets/3490872/4f06de28-0e8d-4d54-ac8d-d900aa2d4817)

github doesn't really seem to properly display this? 🤔

Reply:
So yeah, this PR is a bit "fake": we don't currently deploy the website from the `master` branch, but instead from the `website` branch. Not all logic there has been properly cleaned up yet, and some of the changes there are really not acceptable on `master`; hence the split.

But... these changes in this PR are also part of that branch, so you can see the result [live on the internet](https://error-prone.picnic.tech/compatibility/).